packages/service/src/core/runTest.ts

```diff
@@ -66,7 +63,7 @@ export async function runTest(runTestContext: RunTestContext): Promise<InvokeRes
                 // force retry if test was unsuccesfull
                 // if last retry, return as normal
                 if (!result.success) {
-                    if (retryCount !== runTestContext.maxRetryCount) {
+                    if (attempt <= runTestContext.maxRetryCount) {
```

Comment:
This will trigger on first run regardless of `maxRetryCount`

```suggestion
                    if (attempt >= runTestContext.maxRetryCount) {
```

Reply:
Yeah, we want it to throw an error for all failures _except_ the last run. Since for the last run we need to convert the "error" into a status/result.